### PR TITLE
Jp/french language support

### PR DESCRIPTION
**Pull request by j13panchal**
Added French translation of CCC mobile UI
Added Add CCC job ID to analytics message

## Summary
https://dimagi.atlassian.net/browse/CCCT-338
https://dimagi.atlassian.net/browse/CCCT-340

## Safety Assurance

- [ ] If the PR is high risk, "High Risk" label is set
- [ ] I have confidence that this PR will not introduce a regression for the reasons below
- [ ] Do we need to enhance manual QA test coverage ? If yes, "QA Note" label is set correctly

### CODE REVIEW

File: app/src/org/commcare/CommCareApplication.java [hunk 1]
```diff
@@ -275,9 +279,42 @@ protected void turnOnStrictMode() {
...
 
     @Override
-    public void onConfigurationChanged(Configuration newConfig) {
+    public void onConfigurationChanged(@NonNull Configuration newConfig) {
         super.onConfigurationChanged(newConfig);
         LocalePreferences.saveDeviceLocale(newConfig.locale);
+        updateLocale(this);
+    }
+
+    private void updateLocale(Context context) {
```
Comment (OrangeAndGreen) [inline app/src/org/commcare/CommCareApplication.java:288]: Is this method needed? Seems like the OS properly updates views when the user switches languages, even if I comment this out.

File: app/res/values-fr/strings.xml [hunk 2]
```diff
@@ -161,7 +161,7 @@ License.
     <string name="choose_sound">Choisissez le Son</string>
     <string name="choose_video">Choisissez la Vidéo</string>
     <string name="choose_image">Choisissez l\'Image</string>
-    <string name="selected">"Sélectioné: "</string>
+    <string name="selected" tools:ignore="PrivateResource">"Sélectioné: "</string>
```
Comment (OrangeAndGreen) [inline app/res/values-fr/strings.xml:164]: Curious about why the change here?

**Review by OrangeAndGreen — COMMENTED** [review]
I added a couple questions. Also, I started going through the various screens and generating screenshots to identify issues and/or strings that still need to be translated. I linked the screenshots doc to the Jira ticket. So far I've gone through most of the Connect screens but we still need to look at the ConnectId screens as well.

File: app/res/values-fr/strings.xml (hunk 2 above)
Comment (j13panchal) [reply app/res/values-fr/strings.xml:164]: No other changes were made here. I just resolved the warning that was shown.

File: app/src/org/commcare/CommCareApplication.java (hunk 1 above)
Comment (j13panchal) [reply app/src/org/commcare/CommCareApplication.java:288]: Yes, you're right. The OS automatically updates the view after changing the language from the system settings without needing to set the locale in the code.

Comment (j13panchal) [reply app/src/org/commcare/CommCareApplication.java:288]: Code updated

File: app/src/org/commcare/CommCareApplication.java [hunk 3]
```diff
@@ -275,9 +277,8 @@ protected void turnOnStrictMode() {
     }
 
     @Override
-    public void onConfigurationChanged(Configuration newConfig) {
+    public void onConfigurationChanged(@NonNull Configuration newConfig) {
         super.onConfigurationChanged(newConfig);
-        LocalePreferences.saveDeviceLocale(newConfig.locale);
```
Comment (OrangeAndGreen) [inline app/src/org/commcare/CommCareApplication.java:280]: I think this line may still be necessary for other parts of the code

File: app/src/org/commcare/activities/CommCareSetupActivity.java [hunk 4]
```diff
@@ -888,6 +888,7 @@ private boolean isSingleAppBuild() {
     public void onRequestPermissionsResult(int requestCode,
                                            @NonNull String[] permissions,
                                            @NonNull int[] grantResults) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
```
Comment (OrangeAndGreen) [inline app/src/org/commcare/activities/CommCareSetupActivity.java:891]: Thinking this should be okay but just want to make sure it wasn't left out on purpose since this is traditional CC code. @shubham1g5?

File: app/src/org/commcare/activities/LoginActivityUiController.java [hunk 5]
```diff
@@ -550,6 +550,7 @@ protected void setConnectIdLoginState(boolean useConnectId) {
             text = Localization.get("login.button");
         }
         loginButton.setText(text);
+        connectLoginButton.setText(activity.getString(R.string.connect_button_logged_in));
```
Comment (OrangeAndGreen) [inline app/src/org/commcare/activities/LoginActivityUiController.java:553]: This is set statically in the XML, does it need to be updated here?

Comment (j13panchal) [reply app/src/org/commcare/activities/LoginActivityUiController.java:553]: I included these lines because when a user changes the language from the system settings while in the app, the strings don't translate correctly when they return to the app. That's why I added this code.

File: app/src/org/commcare/activities/CommCareSetupActivity.java (hunk 4 above)
Comment (j13panchal) [reply app/src/org/commcare/activities/CommCareSetupActivity.java:891]: I don't think this will cause any issues in the future. The code was showing an error, which is why I resolved it. However, if you still want to remove this line, I can do that.

File: app/src/org/commcare/CommCareApplication.java (hunk 3 above)
Comment (j13panchal) [reply app/src/org/commcare/CommCareApplication.java:280]: Yes, that's correct. This line is already present in the onCreate method, which is why I removed it from here.

File: app/src/org/commcare/activities/LoginActivityUiController.java (hunk 5 above)
Comment (OrangeAndGreen) [reply app/src/org/commcare/activities/LoginActivityUiController.java:553]: Ah, yeah I see it happening too, not sure why. Add a comment above explaining why we're doing it just so we don't forget.

File: app/src/org/commcare/CommCareApplication.java (hunk 3 above)
Comment (j13panchal) [reply app/src/org/commcare/CommCareApplication.java:280]: Yes, LocalePreferences should update correctly after changing the language while the app is running.

I have already tested this scenario.

File: app/src/org/commcare/activities/LoginActivityUiController.java (hunk 5 above)
Comment (j13panchal) [reply app/src/org/commcare/activities/LoginActivityUiController.java:553]: Okay

**Review by OrangeAndGreen — APPROVED** [review]
